generate_analytical_dataset.py

```diff
@@ -61,18 +79,28 @@ def create_hive_on_published(parquet_location, published_database_name, spark, t
     spark.sql(src_hive_select_query).show()
 
 
-def persist_parquet(S3_PUBLISH_BUCKET, table_to_process, values):
+def persist_parquet(s3_publish_bucket, collection_name, values):
     row = Row("val")
     datadf = values.map(row).toDF()
     datadf.show()
-    adg_parquet_name = table_to_process + "." + "parquet"
+    adg_parquet_name = collection_name + "." + "parquet"
     parquet_location = "s3://%s/${file_location}/%s" % (
-        S3_PUBLISH_BUCKET,
-        adg_parquet_name,
+        s3_publish_bucket,
+        collection_name,
+        adg_parquet_name
     )
     datadf.write.mode("overwrite").parquet(parquet_location)
     return parquet_location
 
+def tag_objects(s3_publish_bucket, prefix, collection_name, collections_all):
+    session = boto3.session.Session()
+    client_s3 = session.client(service_name='s3')
+    default_value = 'default'
+    if collections_all[collection_name] is None or collections_all[collection_name] == '':
```

Comment:
This might lead to key error if the key is not found .Can collections_all.get(key).else(default value) be used here?